common/src/main/java/org/corfudb/common/metrics/micrometer/JVMMetrics.java

```diff
@@ -0,0 +1,43 @@
+package org.corfudb.common.metrics.micrometer;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.Optional;
+import lombok.Data;
+import lombok.Getter;
+import sun.management.HotspotRuntimeMBean;
+import sun.management.ManagementFactoryHelper;
+
+/** Provides JVM level metrics. */
+
+public final class JVMMetrics {
+
+    @Getter(lazy=true)
+    private final static HotspotRuntimeMBean runtimeMBean = ManagementFactoryHelper.getHotspotRuntimeMBean();
+
+    @Data
+    private static class SafePointStats {
+        long safepointTime;
+        long safepointCount;
+    }
+
+    public static void register(Optional<MeterRegistry> metricsRegistry) {
+
+        if (metricsRegistry.isPresent()) {
```

Comment:
maybe we should use micrometer's JVM metrics? https://micrometer.io/docs/ref/jvm

Reply:
I checked micrometer's jvm metrics, but they don't collect TotalSafepointTime. Maybe because its a "private" API.